pkg/provisioners/managers/cluster/provisioner.go

```diff
@@ -316,14 +316,28 @@ func (p *Provisioner) Deprovision(ctx context.Context) error {
 }
 
 func (p *Provisioner) tags(pool *unikornv1.ComputeClusterWorkloadPoolsPoolSpec) *coreapi.TagList {
-	return &coreapi.TagList{
-		coreapi.Tag{
-			Name:  coreconstants.ComputeClusterLabel,
-			Value: p.cluster.Name,
-		},
-		coreapi.Tag{
-			Name:  WorkloadPoolLabel,
-			Value: pool.Name,
-		},
+	tagSet := map[string]string{
+		coreconstants.ComputeClusterLabel: p.cluster.Name,
+		WorkloadPoolLabel:                 pool.Name,
+	}
+
+	// Propagate any additional tags from the cluster's spec, if present
+	if p.cluster.Spec.Tags != nil {
```

Comment:
This nil check is redundant as per https://go.dev/ref/spec#RangeClause (see point 1).